src/actions/exchangeActions.js

```diff
@@ -379,3 +384,56 @@ export const getExchangeSupportedAssetsAction = (callback?: () => void) => {
     dispatch(saveDbAction('exchangeSupportedAssets', { exchangeSupportedAssets }, true));
   };
 };
+
+export const getWbtcFeesAction = () => (dispatch: Dispatch) => {
+  fetch('https://lightnode-mainnet.herokuapp.com', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      id: 67,
+      jsonrpc: '2.0',
+      method: 'ren_queryFees',
+      params: {},
+    }),
+  })
+    .then(res => res.json())
+    .then(res => {
+      dispatch({ type: SET_WBTC_FEES, payload: res.result });
+    })
+    .catch(e => reportErrorLog('Failed to fetch WBTC fees', e));
+};
+
+export const getWbtcGatewayAddressAction = (params: WBTCGatewayAddressParams) =>
+  async (dispatch: Dispatch, getState: GetState, api: SDKWrapper): Promise<WBTCGatewayAddressResponse | null> => {
+    const { user: { data: user } } = getState();
+    const walletId = user?.walletId || '';
+    const gatewayAddressResponse = await api.getWbtcCafeGatewayAddress({ ...params, walletId });
+    return gatewayAddressResponse;
+  };
+
+export const addWbtcPendingTxAction = (tx: PendingWBTCTransaction) => (dispatch: Dispatch, getState: GetState) => {
+  dispatch({
+    type: ADD_WBTC_PENDING_TRANSACTION,
+    payload: tx,
+  });
+  const { exchange: { data: { pendingWbtcTransactions } } } = getState();
+  dispatch(saveDbAction('pendingWbtcTransactions', { pendingWbtcTransactions }, true));
+};
+
+export const setWbtcPendingTxsAction = (pendingWbtcTransactions: PendingWBTCTransaction[]) => (dispatch: Dispatch) => {
+  dispatch({
+    type: SET_WBTC_PENDING_TRANSACTIONS,
+    payload: pendingWbtcTransactions,
+  });
+  dispatch(saveDbAction('pendingWbtcTransactions', { pendingWbtcTransactions }, true));
+};
+
+// for each new WBTC.Cafe tx detected, remove 1 pending
+export const removeWbtcPendingTxsAction = (quantity: number) => (dispatch: Dispatch, getState: GetState) => {
+  if (quantity < 1) return;
+  const { exchange: { data: { pendingWbtcTransactions: pending } } } = getState();
+  if (pending.length >= quantity) {
+    const updatedPendingTransactions = pending.slice(0, pending.length - quantity);
```

Comment:
What if we have several pending transactions? Do we have a way to know which one needs to be removed? Especially since I can add a transaction without actually sending anything

Reply:
We can't tell this, there's no way we can match a mocked pending tx object to the actual transaction. That's why, for n new transactions, we're deleting n pending txs. There's also no way we can verify if the user actually sent the BTC, so once they press on "I've sent BTC", a mocked pending tx is generated (and it's deleted when a WBTC.Cafe tx arrives or after 12 hrs - whichever happens first).

That said, in the code snippet above we're removing pending txs from the end of their array, while we should delete from the beginning - I'll change that.